### DISCUSSION TITLE
Examples on Typescript

### DISCUSSION THREAD

**Discussion by mayorovad** [maxGraph/maxGraph#130] [Ideas]
💡 Is there any plans on porting examples to Typescript? Because now it's just a little strange to fix some typos and inconsistencies in js stories (like #125 `perimeter`) when library itself written in Typescript. 

I tried to write story in Storybook on Typescript with no success - there are sophisticated ways of configuring stories build and I eventually gave up 😿 . In addition, I don't see any benefits from Storybook in current state, besides search box and organizing stories, which can be made easily without it.

And if we have plans on dedicated page (maybe on github.io) it will be benefitial to have own example page, so users not forced to run Storybook. Or, alternatively, there is still posibility to setup Storybook and push it to github.io or some page.

**Comment by tbouffard** [comment]
Hi
about publishing the examples to GitHub Pages, yes, this will be great because it is a pain to have to build the whole project just to see the examples. The API docs could be published as well.
We mentioned doing it in the past, but no one had time to implement it. I will see if I can work on this in the next weeks.

About the JS typo and inconsistencies, this is clearly a pain and it occurred several times in the past. It would clearly better to have examples written in TypeScript to catch such issues easily.

I have done some archeology to try to understand why we choose Storybook, but I didn't find a clear answer
- 14 april 2021 https://github.com/maxGraph/maxGraph/discussions/20#discussioncomment-607441
- 12 sept 2021 https://github.com/maxGraph/maxGraph/discussions/51#discussioncomment-1312739

**Reply by tbouffard** [reply]
New issues with JS inconsistencies were recently detected. See  #164

**Comment by tbouffard** [comment]
I have started working on bumping examples to storybook v7 in #230.
This will allow to publish the stories to GitHub Pages and write the stories using TypeScript.
This should be completed by the end of August.
